### PR TITLE
NIFI-8386: Ensure that we set (and merge) bulletins when creating Aff…

### PR DESCRIPTION
**Pull request by markap14**
…ectedComponent entities and ControllerService Reference entities

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by rfellows — APPROVED** [review]
LGTM 👍

**Comment by mcgilman** [comment]
Reviewing....

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/dto/DtoFactory.java [hunk 1]
```diff
@@ -2857,7 +2857,14 @@ public AffectedComponentEntity createAffectedComponentEntity(final ComponentNode
             }
         }
 
-        return entityFactory.createAffectedComponentEntity(affectedComponent, revision, permissions, groupNameDto);
+        final List<BulletinDTO> bulletins = createBulletins(componentNode);
+        return entityFactory.createAffectedComponentEntity(affectedComponent, revision, permissions, groupNameDto, bulletins);
+    }
+
+    private List<BulletinDTO> createBulletins(final ComponentNode componentNode) {
+        final PermissionsDTO permissions = createPermissionsDto(componentNode);
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/dto/DtoFactory.java:2865]: This is unused.

**Review by mcgilman — COMMENTED** [review]
In non clustered mode the bulletins could be nulled out. Returning the filtered bulletin entity is only really needed for the bulletin board page. This could be consistent with merging when clustered and provide a smaller response payload.

**Comment by mcgilman** [comment]
Thanks @markap14! This have been merged to master.

**Pull request closed without merging**